### PR TITLE
IE C API support for the remote context/blob

### PR DESCRIPTION
**Pull request by xiaoxial**

### CODE REVIEW

**Comment by lxx02** [comment]
@ilya-lavrenov Please have a code review, thanks.

**Comment by ilya-lavrenov** [comment]
@lxx02 could you please commit changes from intel email?

File: inference-engine/ie_bridges/c/include/c_api/ie_c_api.h [hunk 1]
```diff
@@ -302,6 +303,47 @@ typedef struct ie_available_devices {
...
+/**
+ * @struct shared_blob_param_t
+ * @brief The structure contains shared memory blob handle user-supplied and shared blob type paramters
+ * for creating shared blob.
+ */
+typedef struct shared_blob_param {
+    void *blob_handle;  // memory handle user-supplied, such as cl_mem, VASurfaceID, ID3D11Buffer, ID3D11Texture2D,
+    shared_blob_type_e type;
+    uint32_t plane;
+}shared_blob_param_t;
```
Comment (vinograd47) [inline inference-engine/ie_bridges/c/include/c_api/ie_c_api.h:345]: All this structs/enums looks like GPU plugin specific.

Comment (vinograd47) [reply inference-engine/ie_bridges/c/include/c_api/ie_c_api.h:345]: I would expect something like this:

```
struct param {
    int ID;  // or const char* name
    void* value;
};

ie_core_create_context(..., const param* params, int numParams);
```

The C API wrapper should convert the `params` C array to C++ `ParamMap` based on `ID`.

Comment (vinograd47) [reply inference-engine/ie_bridges/c/include/c_api/ie_c_api.h:345]: Or even introduce separate C API for `ParamMap`:

```
struct param_map_t;
param_map_t* create_param_map();
void free_param_map(param_map_t* params);
void add_int_param(param_map_t* params, const char* name, int value);
void add_float_param(param_map_t* params, const char* name, float value);
...
```

**Comment by guoyejun** [comment]
hi, any further plan for this PR, it is helpful and we want the c api, thanks.

**Comment by ilya-lavrenov** [comment]
Will be implemented in OpenVINO C API 2.0
CC @riverlijunjie

**Pull request closed without merging**